File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        
        int n=height.size();
        
        int max_area=INT_MIN,i=0,j=n-1;
        
        while(i<j) 
        {
            int h=min(height[i],height[j]);
            
            max_area=max(max_area,h*(j-i));
            
            if(height[i]<=h)
                i++;
            if(height[j]<=h)
                j--;
        }
        return max_area;
        
    }
};

File: 11-container-with-most-water/11-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left=0;
        int right=height.length-1;
        int maxArea=0;
        int lh, rh;
        while(left<right)
        {
            lh=height[left];
            rh=height[right];
            int minimum= Math.min(lh,rh);
            maxArea= Math.max(maxArea, minimum*(right-left));
            if(lh<rh)
                left++;
            else
                right--;
        }
        return maxArea;
    }
}